### PR TITLE
[SYCLomatic] Support migration of 36 SIMD math APIs.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

File: clang/runtime/dpct-rt/include/dpct/math.hpp [hunk 1]
```diff
@@ -655,19 +666,37 @@ struct average {
...
   auto v3 = v1.as<VecT>();
   auto v4 =
       detail::vectorized_binary<VecT, BinaryOperation>()(v2, v3, binary_op);
+  if (need_relu)
+    v4 = relu(v4);
   v0 = v4.template as<sycl::vec<unsigned, 1>>();
   return v0;
 }
 
+/// TODO:.
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/dpct/math.hpp]: Let's remove this.

File: clang/runtime/dpct-rt/include/dpct/math.hpp [hunk 2]
```diff
@@ -750,6 +779,24 @@ inline unsigned vectorized_sum_abs_diff(unsigned a, unsigned b) {
   return sum;
 }
 
+/// TODO:.
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/dpct/math.hpp]: Same as above

File: clang/runtime/dpct-rt/include/dpct/math.hpp [hunk 3]
```diff
@@ -750,6 +779,24 @@ inline unsigned vectorized_sum_abs_diff(unsigned a, unsigned b) {
...
+template <typename VecT, class BinaryOperation1, class BinaryOperation2>
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/dpct/math.hpp]: I think we should all use `class` or `typename`.

File: clang/runtime/dpct-rt/include/dpct/math.hpp [hunk 4]
```diff
@@ -655,19 +666,37 @@ struct average {
...
   auto v4 =
       detail::vectorized_binary<VecT, BinaryOperation>()(v2, v3, binary_op);
+  if (need_relu)
+    v4 = relu(v4);
   v0 = v4.template as<sycl::vec<unsigned, 1>>();
   return v0;
 }
 
+/// TODO:.
+template <typename T, class BinaryOperation>
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/dpct/math.hpp]: I think we should all use class or typename.

Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/math.hpp]: done

File: clang/runtime/dpct-rt/include/dpct/math.hpp (hunk 3 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/math.hpp]: done

File: clang/runtime/dpct-rt/include/dpct/math.hpp (hunk 1 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/math.hpp]: done

File: clang/runtime/dpct-rt/include/dpct/math.hpp (hunk 2 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/math.hpp]: done

**Review by yihwang-nv — APPROVED** [review]
LGTM

**Comment by tangjj11** [comment]
https://github.com/intel/llvm/pull/15550 merged.

**Review by zhiweij1 — APPROVED** [review]

**Review by ziranzha — APPROVED** [review]

**Pull request merged**